### DISCUSSION TITLE
Lesson 13: front_end/src/deployments/map.json file

### DISCUSSION THREAD

**Discussion by toddabraham** [smartcontractkit/full-blockchain-solidity-course-py#262] [Q&A] [answered]
Hi there,

I may have missed something, but are we supposed to create the map.json file, or is it automatically created and populated after a deployment?

![image](https://user-images.githubusercontent.com/26919212/138711786-40d576e7-a618-4185-9265-6730a2a449d4.png)

**Comment by cromewar** [comment, marked as answer]
This is a tricky one, but let me try to solve the question.

When you compile a project on Brownie, Map.json gets generated automatically saving the recent deployments yo made, the info of the networks comes directly from `brownie-config.yaml`. As JavaScripts cannot understand things outside the src you send that information at deployment with the `update_front_end` function which sends a Json File.

![image](https://user-images.githubusercontent.com/20199180/138730229-4acf5386-1152-4f58-bc78-6c5c29fb789a.png)


Also you need the build folder so you send it using `copy_folders_to_frontend` function

![image](https://user-images.githubusercontent.com/20199180/138730320-1564feb8-1608-4bf5-8adc-c059219ea893.png)
![image](https://user-images.githubusercontent.com/20199180/138730453-3f48d81e-9808-48d2-8795-1f26fab49fcb.png)

As you can see, `map.json`comes from brownie compilation.

Hope this might solve your question.

**Reply by toddabraham** [reply]
Thanks again @cromewar .  I had done all you suggested, the issue was in my .env I hadn't included my ETHER_SCAN API token ID.

**Reply by cromewar** [reply]
I'm glad you solved it, feel free to make any another question.
